src/ScalarMessaging.ts

```diff
@@ -322,6 +346,35 @@ function inviteUser(event: MessageEvent<any>, roomId: string, userId: string): v
     });
 }
 
+function kickUser(event: MessageEvent<any>, roomId: string, userId: string): void {
+    logger.log(`Received request to kick ${userId} from room ${roomId}`);
+    const client = MatrixClientPeg.get();
+    if (!client) {
+        sendError(event, _t('You need to be logged in.'));
+        return;
+    }
+    const room = client.getRoom(roomId);
+    if (room) {
+        // if they are already not in the room we can resolve immediately.
+        const member = room.getMember(userId);
+        if (!member || (member && member.membership === "leave")) {
+            sendResponse(event, {
+                success: true,
+            });
+            return;
+        }
+    }
+
+    const reason = event.data.reason;
+    client.kick(roomId, userId, reason).then(function() {
+        sendResponse(event, {
+            success: true,
+        });
+    }, function(err) {
+        sendError(event, _t('You need to be able to kick users to do that.'), err);
+    });
```

Comment:
```suggestion
    client.kick(roomId, userId, reason).then(() => {
        sendResponse(event, {
            success: true,
        });
    }).catch((err) => {
        sendError(event, _t("You need to be able to kick users to do that."), err);
    });
```

arrow functions, then/catch, etc for modern code style